### PR TITLE
[api] Remove dependency of library on Vernacexpr.

### PR DESCRIPTION
**Pull request by ejgallego**
Morally, `library` should not depend on the vernacular
definition. This will also create problems when trying to modularize
the codebase due to the cycle [vernacs depend for example on
constrexprs].

The fix is fortunately easy.

### CODE REVIEW

File: intf/vernacexpr.ml
Comment (SkySkimmer) [inline intf/vernacexpr.ml]: Why not keep these as deprecated aliases?

Comment (ejgallego) [reply intf/vernacexpr.ml]: well-spotted merci.

**Comment by ejgallego** [comment]
I guess this should be driven the maintainers of `library` @silene @gares

**Review by herbelin — APPROVED** [review]
Most reasonable and welcome.

**Comment by ejgallego** [comment]
Pinging maintainers of `vernac` @mattam82 @maximedenes and of intf @ppedrot @letouzey in case they'd like to drive this one.

**Comment by ejgallego** [comment]
8 days open and no action from maintainers, assigning secondary ones for review as a first step.

**Review by ppedrot — APPROVED** [review]
I dont' have a particular comment on that one. If that can indeed simplify dependencies so easily, then that's good to me.

**Comment by ppedrot** [comment]
The changes are just minor code moves, so I don't think this really requires reviews by other owners. I'm letting you a little time to comment on that one though, just in case.

**Comment by ejgallego** [comment]
I agree with you, this patch is just a nit or technical artifact.

**Comment by SkySkimmer** [comment]
Since the deprecated aliases were put back this doesn't need an overlay.

**Comment by ejgallego** [comment]
Still leaving the overlay in place as it is good to remove the warning.

**Pull request merged**
